addons/l10n_it_edi/wizard/account_move_send.py

```diff
@@ -1,12 +1,216 @@
+        """ Don't allow the checkbox to be clicked if there already is a PDF,
+            this may lead to inconsistencies between the PDF and the EDI XML
+        """
+        for wizard in self:
+            already_has_pdf = any(wizard.move_ids.mapped("invoice_pdf_report_id"))
+            already_has_xml = any([x._l10n_it_edi_check_processable() for x in wizard.move_ids.mapped("attachment_ids")])
+            wizard.l10n_it_edi_readonly = wizard.l10n_it_edi_warning or already_has_pdf or already_has_xml
+
+    @api.depends('l10n_it_edi_readonly', 'enable_l10n_it_edi')
+    def _compute_l10n_it_edi_checkbox(self):
+        for wizard in self:
+            wizard.l10n_it_edi_checkbox = wizard.enable_l10n_it_edi and not wizard.l10n_it_edi_readonly
+
+    @api.constrains('l10n_it_edi_checkbox')
+    def _check_l10n_it_edi_checkbox(self):
+        for wizard in self:
+            if wizard.l10n_it_edi_checkbox and not wizard.wizard._get_proxy_user():
+                raise UserError(_("You must accept the terms and conditions in the Settings to use the IT EDI."))
+
+    # -------------------------------------------------------------------------
+    # ATTACHMENTS
+    # -------------------------------------------------------------------------
+
+    def _get_mail_attachment_from_doc(self, move):
+        # OVERRIDES 'account'
+        # Don't send EDI XMLs via mail
+        if self.l10n_it_edi_checkbox or move.l10n_it_edi_state:
             return []
-        return super()._get_mail_attachment_from_doc(doc)
+        return super()._get_mail_attachment_from_doc(move)
+
+    # -------------------------------------------------------------------------
+    # BUSINESS ACTIONS
+    # -------------------------------------------------------------------------
+
+    def _hook_invoice_document_before_pdf_report_render(self, invoice, invoice_data):
+        # EXTENDS 'account'
+        super()._hook_invoice_document_before_pdf_report_render(invoice, invoice_data)
+        if self.l10n_it_edi_checkbox:
+            if (errors := invoice._l10n_it_edi_export_data_check()):
+                error_str = _("Errors occured while creating the e-invoice file.")
+                error_str += "\n- " + "\n- ".join(errors)
+                invoice_data['error'] = error_str
+            else:
+                invoice_data['l10n_it_edi_content'] = self.env['l10n_it_edi.export']._l10n_it_edi_export(invoice)
+                invoice_data['l10n_it_edi_filename'] = self.env['ir.attachment']._l10n_it_edi_generate_filename(invoice.company_id)
+
+    def action_send_and_print(self, from_cron=False):
+        # EXTENDS 'account'
+        self.ensure_one()
+
+        # Mark the moves as 'being_sent'
+        if self.l10n_it_edi_checkbox:
+            for move in self.move_ids:
+                if move.l10n_it_edi_state in (False, 'rejected'):
+                    move.l10n_it_edi_state = 'being_sent'
+
+        return super().action_send_and_print(from_cron)
```

Comment:
You shoudn't extend this method. All status update should be done in `_call_web_service` with a commit between each call